app/services/proxy_status/base.rb

```diff
@@ -0,0 +1,40 @@
+module ProxyStatus
+  class Base
+    def initialize(proxy, opts = {}, api_class_name = 'Version')
+      @proxy = proxy
+      @cache_duration = opts[:cache_duration] || 3.minutes
+      @api_class_name = api_class_name
+    end
+
+    def revoke_cache!
+      # As memcached does not support delete_matched, we need to delete each
+      Rails.cache.delete(cache_key)
+    end
+
+    def cache_key
+      raise NotImplementedError _('Implement in subclasses')
```

Comment:
This is missing a comma between Error and the string, Also the string should say something like `cache_key not implemented for ProxyStatus#{self.class}` or something similar, so that it stands out on logs